PROBEZ

```diff
@@ -1,44 +1,56 @@
-(AUTO TOOL SET)
-(T99 is Probe)
+(PROBEZ)
+(Description: Probe in Z from the above stock)
+(Initial Coding: Justin Gray)
+(Modified 1/14/2024: Joshua Smith)
+(Simplified WCS extended number math, added probe error checking, added probe distance argument,removed air blast)
+(Argument A -> #1 is the work coordinate system to store offsets in. The format for extended G54 offsets would be a period followed by the offset(G54.5))
+(Argument B -> #2 is the desired probing distance and should be negative)
+
+(load probe config)
+G65 "PROBECONFIG"
+
+(important local variables)
+#100 = @100             (TOOL NUMBER PROVIDED BY PROBECONFIG MACRO)
+#110 = @103             (FEED SPEED PROVIDED BY PROBECONFIG MACRO)
+#111 = @104             (FAST PROBE SPEED PROVIDED BY PROBECONFIG MACRO)
+#112 = @105             (SLOW PROBE SPEED PROVIDED BY PROBECONFIG MACRO)
+#105 = @107             (TOOL LENGTH PROVIDED BY PROBECONFIG MACRO)
+#108 = @108             (PROBE BACKOFF DISTANCE PROVIDED BY PROBECONFIG MACRO)
+
+(CALCULATE EXTENDED WCS NUMBER)
+(FIX is a round down function and MOD is modulo)
+#114 = MOD[#1,FIX[#1]]*10
+
+(Check that the probe distance is negative)
+IF[#2<0]
+
+(Probe z the desired distance and at fast feed)
+G31 G91 P2 Z[#2] F#111
+
+ (Check that the probe has triggered)
+ IF[R_SKIP[0,1] == 1]
+  G91 G01 Z+[#108]                     (back off 1/4 the probing distance)
+  G31 G91 P2 Z[#2] F#112               (Probe X the desired distance at slow feed)
+  #104=R_SKIP[0,203]                   (GET MACHINE X COORDINATE)
+  G91 G01 Z+[#108]                     (back off 1/4 the probing distance)
+  #106 = [#104-#105]                   (current position - probe length)
+
+  (STORE X OFFSET FOR WCS ZERO)
+  IF [#114 < 1]  
+    W_G53G59_COOR[0,#1,3,#106]
+  ELSE
+    W_G54EXP_COOR[0,#114,3,#106]
+  END_IF
+
+ ELSE
+  MSG_OK["PROBEZ", "ERROR:  FAILED TO PROBE PART WITHIN SPECIFIED DISTANCE PRESS CYCLE STOP AND THEN RESET",""]
```

Comment:
Should we change these to actual errors on the control? rather than just messages?